base/uk.ac.stfc.isis.ibex.epics.switching/src/uk/ac/stfc/isis/ibex/epics/switching/InstrumentSwitchers.java

```diff
@@ -88,9 +98,18 @@ public Switcher getObservableSwitcher(OnInstrumentSwitch switchType) {
     }
 
     public void setInstrument(InstrumentInfo instrument) {
+    	switching = true;
         instance.nothingSwitcher.switchInstrument(instrument);
         instance.closingSwitcher.switchInstrument(instrument);
         instance.observablePrefixChangingSwitcher.switchInstrument(instrument);
         instance.writablePrefixChangingSwitcher.switchInstrument(instrument);
+        // This is to prevent instrument from being switched too often
+        ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+        executorService.schedule(new Runnable() {
+			@Override
+			public void run() {
+		        switching = false;
```

Comment:
This is more of a question than a comment, but how/where does this get set back to `true`?